test/extended/networking/bonds.go

```diff
@@ -0,0 +1,79 @@
+package networking
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	nadtypes "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	kapiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+)
+
+var _ = g.Describe("[sig-network][Feature:bond]", func() {
+	oc := exutil.NewCLI("bond")
+	f := oc.KubeFramework()
+
+	g.It("pod with bond interface should be created ", func() {
+		namespace := f.Namespace.Name
+		podName1, podName2 := "pod1", "pod2"
+		bondnad1, bondnad2 := "bondnad1", "bondnad2"
+
+		g.By("creating network attachment definitions")
+		o.Expect(createNetworkAttachmentDefinition(oc.AdminConfig(), namespace, "macvlannad", `{"cniVersion":"0.4.0","name":"macvlan-nad","plugins":[{ "type": "macvlan"}]}`)).NotTo(o.HaveOccurred(), "unable to create macvlan network-attachment-definition")
+		o.Expect(createBondNAD(oc.AdminConfig(), namespace, bondnad1, "10.0.0.10/24", "net1", "net2")).NotTo(o.HaveOccurred(), "unable to create bond network-attachment-definition")
+		o.Expect(createBondNAD(oc.AdminConfig(), namespace, bondnad2, "10.0.0.11/24", "net1", "net2")).NotTo(o.HaveOccurred(), "unable to create bond network-attachment-definition")
+
+		g.By("creating first pod and checking results")
+		exutil.CreateExecPodOrFail(f.ClientSet, namespace, podName1, func(pod *kapiv1.Pod) {
+			macVlanAnnotation := fmt.Sprintf("%s/%s", namespace, "macvlannad")
+			bondAnnotation := fmt.Sprintf("%s/%s@%s", namespace, bondnad1, bondnad1)
+			pod.ObjectMeta.Annotations = map[string]string{"k8s.v1.cni.cncf.io/networks": fmt.Sprintf("%s, %s, %s", macVlanAnnotation, macVlanAnnotation, bondAnnotation)}
+		})
+		pod, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(context.TODO(), podName1, metav1.GetOptions{})
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		networkStatusString, ok := pod.Annotations["k8s.v1.cni.cncf.io/network-status"]
+		o.Expect(ok).To(o.BeTrue())
+		o.Expect(networkStatusString).ToNot(o.BeNil())
+		networkStatuses := []nadtypes.NetworkStatus{}
+		o.Expect(json.Unmarshal([]byte(networkStatusString), &networkStatuses)).ToNot(o.HaveOccurred())
+		o.Expect(networkStatuses).To(o.HaveLen(4))
+		o.Expect(networkStatuses[3].Interface).To(o.Equal(bondnad1))
+		o.Expect(networkStatuses[3].Name).To(o.Equal(fmt.Sprintf("%s/%s", namespace, bondnad1)))
```

Comment:
not sure if checking the struct isn't more readable. Something like:

```golang
o.Expect(networkStatuses[3]).To(
    Equal(
        NetworkStatus{
            Interface: bondnad1, 
            Name: fmt.Sprintf("%s/%s", namespace, bondnad1),
        },
))
```

Totally opinionated, feel free to ignore.

Reply:
tbh if the line is short spreading it across multiple lines does not help readability
I would leave it as is